prover/Cargo.toml

```diff
@@ -17,7 +17,8 @@ edition.workspace = true
 async = ["winter-maybe-async/async"]
 concurrent = ["processor/concurrent", "std", "winter-prover/concurrent"]
 default = ["std"]
-metal = ["dep:miden-gpu", "dep:elsa", "dep:pollster", "concurrent", "std"]
+metal = ["dep:miden-gpu", "miden-gpu/metal", "dep:elsa", "dep:pollster", "concurrent", "std"]
+cuda = ["dep:miden-gpu", "miden-gpu/cuda", "std"]
```

Comment:
A couple of small comments:

- For `metal`, we can probably get rid of `std` requirement as it is implied by `concurrent`.
- For `cuda`, is there a reason not to add `concurrent`? If we add that, we could get rid of `std` for similar reason as in the previous point.

Also question: do we actually need `dep:miden-gpu` feature listed, I thought it would be implied by `miden-gpu/metal` and `miden-gpu/cuda`. But maybe not?

Reply:
You are completely right. During development it wouldn't compile for me without "dep:miden-gpu" but it seems to work fine now. Fixed.